bundles/EcommerceFrameworkBundle/VoucherService/TokenManager/Pattern.php

```diff
@@ -541,14 +541,25 @@ public function prepareConfigurationView(&$viewParamsBag, $params)
 
         /** @var PaginatorInterface $paginator */
         $paginator = \Pimcore::getContainer()->get(\Knp\Component\Pager\PaginatorInterface::class);
+
+        $page = (int)($params['page'] ?? 1);
+        $perPage = (int)($params['tokensPerPage'] ?? 25);
+        
+        $total = count($tokens);
+        
+        $availablePages = (int) ceil($total / $perPage);
+        if ($page > $availablePages) {
+            $page = $availablePages;
+        }
```

Comment:
```suggestion
        $page = min($page, (int) ceil($total / $perPage));
```
should also work? since we are looking for available page with min value

Reply:
yes, lgtm even though i feel that with `$availablePages` is a little more clear to understand what's going on